components/raftstore/src/store/fsm/apply.rs

```diff
@@ -3764,9 +3792,36 @@ where
         // queued inside both queue of control fsm and normal fsm, which can reorder
         // messages.
         let (sender, apply_fsm) = ApplyFsm::from_registration(reg);
-        let mailbox = BasicMailbox::new(sender, apply_fsm);
+        let mailbox = BasicMailbox::new(sender, apply_fsm, self.state_cnt().clone());
         self.register(region_id, mailbox);
     }
+
+    pub fn register(&self, region_id: u64, mailbox: BasicMailbox<ApplyFsm<EK>>) {
+        self.router.register(region_id, mailbox);
+        self.update_router_trace();
+    }
+
+    pub fn register_all(&self, mailboxes: Vec<(u64, BasicMailbox<ApplyFsm<EK>>)>) {
+        self.router.register_all(mailboxes);
+        self.update_router_trace();
+    }
+
+    pub fn close(&self, region_id: u64) {
+        self.router.close(region_id);
+        self.update_router_trace();
+    }
+
+    fn update_router_trace(&self) {
+        let router_trace = self.router.trace();
+        RAFTSTORE_MEM_TRACE
```

Comment:
How about extracting a function so that we don't need to write it multiple times?

Reply:
I think there are 4 times that we do the reset operation.

Reply:
it's for different two routers.

Reply:
We still can provide a function and use a parameter to distinguish them. But It's ok for me with the current implementation.

Reply:
Changed them to different static variables.

Reply:
much better